src/gui/search/searchtab.cpp

```diff
@@ -154,19 +179,122 @@ QStandardItemModel* SearchTab::getCurrentSearchListModel() const
 void SearchTab::setRowColor(int row, QString color)
 {
     m_proxyModel->setDynamicSortFilter(false);
-    for (int i = 0; i < m_proxyModel->columnCount(); ++i) {
+    for (int i = 0; i < m_proxyModel->columnCount(); ++i)
         m_proxyModel->setData(m_proxyModel->index(row, i), QVariant(QColor(color)), Qt::ForegroundRole);
-    }
 
     m_proxyModel->setDynamicSortFilter(true);
 }
 
-QString SearchTab::status() const
+SearchTab::Status SearchTab::status() const
 {
     return m_status;
 }
 
-void SearchTab::setStatus(const QString &value)
+void SearchTab::setStatus(Status value)
 {
     m_status = value;
+    setStatusTip(statusText(value));
+    const int thisTabIndex = m_parent->searchTabs()->indexOf(this);
+    m_parent->searchTabs()->setTabToolTip(thisTabIndex, statusTip());
+    m_parent->searchTabs()->setTabIcon(thisTabIndex, GuiIconProvider::instance()->getIcon(statusIconName(value)));
+}
+
+void SearchTab::updateFilter()
+{
+    using Utils::Misc::SizeUnit;
+    SearchSortModel* filterModel = this->getCurrentSearchListProxy();
+    filterModel->enableNameFilter(filteringMode() == OnlyNames);
+    // we update size and seeds filter parameters in the model even if they are disabled
+    // because we need to read them from the model when search tabs switch
+    filterModel->setSeedsFilter(m_ui->minSeeds->value(), m_ui->maxSeeds->value());
+    filterModel->setSizeFilter(
+        Utils::Misc::sizeInBytes(m_ui->minSize->value(), static_cast<SizeUnit>(m_ui->minSizeUnit->currentIndex())),
+        Utils::Misc::sizeInBytes(m_ui->maxSize->value(), static_cast<SizeUnit>(m_ui->maxSizeUnit->currentIndex())));
+    filterModel->invalidate();
+}
+
+void SearchTab::fillFilterComboBoxes()
+{
+    using namespace Utils::Misc;
+    QStringList unitStrings;
+    unitStrings.append(unitString(SizeUnit::Byte));
+    unitStrings.append(unitString(SizeUnit::KibiByte));
+    unitStrings.append(unitString(SizeUnit::MebiByte));
+    unitStrings.append(unitString(SizeUnit::GibiByte));
+    unitStrings.append(unitString(SizeUnit::TebiByte));
+    unitStrings.append(unitString(SizeUnit::PebiByte));
+    unitStrings.append(unitString(SizeUnit::ExbiByte));
+
+    m_ui->minSizeUnit->clear();
+    m_ui->maxSizeUnit->clear();
+    m_ui->minSizeUnit->addItems(unitStrings);
+    m_ui->maxSizeUnit->addItems(unitStrings);
+
+    m_ui->minSize->setValue(0);
+    m_ui->minSizeUnit->setCurrentIndex(static_cast<int>(SizeUnit::MebiByte));
+
+    m_ui->maxSize->setValue(-1);
+    m_ui->maxSizeUnit->setCurrentIndex(static_cast<int>(SizeUnit::TebiByte));
+
+    m_ui->filterMode->clear();
+
+    QMetaEnum nameFilteringModeEnum =
+        this->metaObject()->enumerator(this->metaObject()->indexOfEnumerator("NameFilteringMode"));
```

Comment:
This is to save enum names instead of combobox index in the config file and, at the same time, to not duplicate those names in the code which does saving/loading. So the config file record reads:
```
FilteringMode=Everywhere
```
instead of 
```
FilteringModeIndex=1
```

I believe that this is better with respect to config file readability and possible further modifications. Perhaps I have to add comment to the enum which would warn that name changes will break config reading.